test/cpp/unit/test_config.cpp

```diff
@@ -0,0 +1,98 @@
+//
+// Created by Jason Mohoney on 4/12/21.
+//
+
+#include <gtest/gtest.h>
+#include <config.h>
+#include <string>
+
+std::string conf_str = std::string(MARIUS_TEST_DIRECTORY) + "/test_configs/default.ini";
+
+
+TEST(TestConfig, TestConfigOnly) {
+    // Test that the default config works properly
+    int num_args = 2;
+    const char* n_argv[] = {"marius_train", conf_str.c_str()};
+    parseConfig(num_args, (char **)n_argv);
+}
+
+TEST(TestConfig, TestCommandLineOverride) {
```

Comment:
Could you add another argument to test here for the general.gpu_ids option. `--general.gpu_ids=0 1 2 3 4 5 6 7`

There is a chance this could fail if our command line parsing doesn't handle the whitespace properly.